moveit_ros/planning/kinematics_plugin_loader/src/kinematics_plugin_loader.cpp

```diff
@@ -69,19 +69,17 @@ class KinematicsPluginLoader::KinematicsLoaderImpl
    * \param search_res
    * \param iksolver_to_tip_links - a map between each ik solver and a vector of custom-specified tip link(s)
    */
-  KinematicsLoaderImpl(const NodeInterfaceSharedPtr& node_interface, const TopicsInterfaceSharedPtr& topics_interface,
-                       const ParametersInterfaceSharedPtr& parameters_interface, const std::string& robot_description,
+  KinematicsLoaderImpl(const node_interface::NodeInterfaceSharedPtr& node_interface,
```

Comment:
Tiny bit of feedback is that if you are going to make a copy of a shared_ptr (to increment the reference count, which you are doing here) it is generally good to make it obvious to your caller that you are doing that through your function signature. The way to do that is to not take a reference but instead take it by value, and then move that value into your local state. Here is a minimal example of what I mean:

```Foo(std::shared_ptr<Node> node) : node_{std::move(node)} {}```

Reply:
Good point. I made the change.